File: src/game/characters/Vegeta.ts
import { BaseFighter } from "../BaseFighter";

export class Vegeta extends BaseFighter {
  readonly id = "vegeta" as const;
  readonly name = "Vegeta";
  readonly greeting = "The prince will not lose.";
  readonly signatureAttack = "Final Flash";
  readonly attackStyle = "beam" as const;
  readonly power = 9500;
  readonly speed = 89;
  readonly defense = 79;
  readonly criticalChance = 0.2;
  readonly baseDamage = 20;
  readonly nativeFacing = "left" as const;
  readonly scale = 0.98;
  readonly palette = {
    skin: "#e9aa78",
    hair: "#11121b",
    primary: "#1c39a7",
    secondary: "#f2f4f7",
    accent: "#e8be42",
    aura: "#9c70ff"
  };
}

File: csharp-reference/Characters/Vegeta.cs
using DragonBallArena.Models;

namespace DragonBallArena.Characters;

public sealed class Vegeta(string? name = null) : BaseFighter(name)
{
    public override string Id => "vegeta";
    public override string DisplayName => "Vegeta";
    public override string Greeting => "The prince will not lose.";
    public override string SignatureAttack => "Final Flash";
    public override string AttackStyle => "beam";
    public override double Scale => 0.98;
    public override FighterStats Stats => new(9500, 89, 79, 0.20, 20);
    public override FighterPalette Palette => new(
        "#e9aa78", "#11121b", "#1c39a7", "#f2f4f7", "#e8be42", "#9c70ff");
}
